src/test/java/org/takes/tk/TkWithHeadersTest.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.tk;
+
+import com.google.common.base.Joiner;
+import java.io.IOException;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.takes.rs.RsPrint;
+
+/**
+ * Test case for {@link org.takes.tk.TkWithHeaders}.
+ * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
+ * @version $Id$
+ * @since 0.9.1
+ */
+public final class TkWithHeadersTest {
+
+    /**
+     * TkWithHeaders can add headers.
+     * @throws java.io.IOException If some problem inside
+     */
+    @Test
+    public void addHeaders() throws IOException {
+        final String host = "Host: www.example.com";
+        final String type = "Content-Type: text/xml";
+        MatcherAssert.assertThat(
+            new RsPrint(
+                new TkWithHeaders(
+                    new TkEmpty(),
+                    host,
+                    type
+                ).with("host", "a.example.com").act()
+            ).print(),
+            Matchers.equalTo(
+                Joiner.on("\r\n").join(
+                    "HTTP/1.1 200 OK",
+                    host,
+                    type,
+                    "host: a.example.com",
+                    "",
```

Comment:
@pinaf Yes, these two empty string necessary, I think. Because org.takes.rs.RsPrint.print appends eol after each header and one eol at the end. Should it add eol after last header and at the end? I don't know.
I can't define it as a bug. I do not have enough knowledge about the project.
Maybe I should create new issue to clarify it?